### ISSUE TITLE
🚀 Feature: Any possible to define dynamic object config with frontend visibility?

### ISSUE THREAD

**Issue by JounQin** [backstage/backstage#17886]
### 🔖 Feature description

```ts
{
  /**
   * @visibility frontend
   * @additionalProperties true
   */
  settings?: Record<string, unknown>;
}
```

### 🎤 Context

I want to get `settings` config in frontend but I only get `{}` actually.

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by Rugvip** [comment]
This is intentional, you only get the individual values that are marked as visible and none of the descendants. You'll need to define a more strict schema for the frontend config.

**Comment by shailahir** [comment]
@JounQin As a workaround, you can expose an endpoint from backend that reads those values from config and return.

**Comment by Rugvip** [comment]
Closing as fixed by #19487
